src/librustc/infer/canonical/mod.rs

```diff
@@ -443,8 +460,19 @@ impl<'tcx> CanonicalVarValues<'tcx> {
                     UnpackedKind::Lifetime(..) => tcx.mk_region(
                         ty::ReLateBound(ty::INNERMOST, ty::BoundRegion::BrAnon(i))
                     ).into(),
-                    UnpackedKind::Const(..) => {
-                        unimplemented!() // FIXME(const_generics)
+                    UnpackedKind::Const(ct) => {
+                        let ty = match ct {
+                            ty::LazyConst::Unevaluated(def_id, _) => {
+                                tcx.type_of(*def_id)
+                            }
+                            ty::LazyConst::Evaluated(ty::Const { ty, .. }) => ty,
```

Comment:
Rebasing is such a pain.

Reply:
Sure, but this PR makes very little sense atm anywhere "lazy" shows up. We'd have to fix it either before or after, and I want to be able to review this PR instead of just skimming it.

Reply:
Ok, I'll create a PR cleaning up the situation

Reply:
#59178 is merged.